### PR TITLE
History Functionality

### PR DESCRIPTION
**Pull request by shraddha-amber**
This PR contains:
1. Add a single video to history
2. Delete a single video from history
3. Clear history

image

### CODE REVIEW

**Review by ragini07 — COMMENTED** [review]
-Good naming convention used
-Liked that you created constants for routes as well
-Preview working as expected
Good to go ahead

**Review by ragini07 — APPROVED** [review]

File: src/pages/SingleVideoPage/index.jsx [hunk 1]
```diff
@@ -3,30 +3,38 @@ import React, { useEffect, useState } from "react";
...
+      const {data, status} = await axios.get(`/api/video/${videoId}`);
+      if (status === 200) setCurrVideo(data.video);
+      const isInHistory = history.find((curr) => curr._id === videoId);
+      if (token && !isInHistory) addToHistory(data.video, token, dataDispatch);
     })();
+
     return () => setCurrVideo({});
   }, [videoId]);
 
+  useEffect(() => {});
```
Comment (manismk) [inline src/pages/SingleVideoPage/index.jsx]: Can remove the empty useEffect If not used

**Review by manismk — APPROVED** [review]
Functionality working as expected.
Good naming convention

File: src/pages/SingleVideoPage/index.jsx (hunk 1 above)
Comment (shraddha-amber) [reply src/pages/SingleVideoPage/index.jsx]: thanks mani, removed it

**Pull request merged**